### ISSUE TITLE
Allow to support all methods in a single route definition

### ISSUE THREAD

**Issue by YerkoPalma** [shipharbor/merry#93]
It would be cool if I could define support for GET, POST, PUT, etc in a single route definition. My specific use case, is that I want to define a REST resource using [rest-parser](https://github.com/karissa/node-rest-parser) which can define support for all methods with the [dispatch method](https://github.com/karissa/node-rest-parser#restparserdispatchreq-opts-cb). I think something like this would be fine:

```js
var merry = require('merry')
var app = merry()
var RestParser = require('rest-parser')
var Book = require('./book.js')

app.route('/', function (req, res, ctx) {  
  var id = parseInt(opts.params.id) || opts.params.id
  REST.dispatch(req, { id: id }, function (err, data) {
    ctx.log.info('oh hey, a request here')
    res.end(JSON.stringify(data))
  })
})
```

In other words, when the first argument is a string, and the second argument is a function, the handler would cover all the http methods. The `'default'` route definition would be an exception.

**Comment by yoshuawuyts** [comment]
Think being explicit might be better. E.g. `'*'`. I do see your use case
tho, think it's reasonable (:

**Comment by YerkoPalma** [comment]
Yeah, though about that option too. I'm ok with the `'*'` call, but I'm not sure if that change should be in merry or in server-router

**Comment by yoshuawuyts** [comment]
I think server-router right? PR welcome!

**Comment by lrlna** [comment]
Yea, I think we gotta fix that in server-router.

**Comment by YerkoPalma** [comment]
closed in version 5.3.0 :tada:
